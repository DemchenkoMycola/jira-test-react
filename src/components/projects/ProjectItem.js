import React from 'react';
import { Link } from 'react-router-dom';
import ProjectDetails from './ProjectDetails'

class ProjectItem extends React.Component {

    constructor(props){
        super(props);

        this.state = {
            showDetails: false
        }

        this.showDetails = this.showDetails.bind(this);
        this.deleteProject = this.deleteProject.bind(this);
    }    

    showDetails(){
        this.setState({
            showDetails: !this.state.showDetails
        })
    }

    deleteProject(){
        let id = this.props.project.id;
        this.props.deleteProject(id)
    }

    render(){

        let {id, title} = this.props.project;

        return(
            <div>
                <div className="project_item"> 
                    <div>{title}</div>
                    <button 
                        onClick={this.showDetails}>
                        {
                            !this.state.showDetails ?
                            'show details' : 'hide details'
                        }
                    </button>
                    <Link to={`/issues/${id}`}>to project</Link>
                    <button 
                        onClick={this.deleteProject}
                        className="del_button"
                        >del</button>
                </div>
                {
                    this.state.showDetails ?
                    <ProjectDetails project={this.props.project} />
                    : null
                }
            </div>
        )
    }
}

export default ProjectItem;