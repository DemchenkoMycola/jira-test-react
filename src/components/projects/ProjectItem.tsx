import React from 'react';
import { Link } from 'react-router-dom';
import { FaRegTrashAlt } from 'react-icons/fa';

interface Project {
    id: string,
    title: string
}

interface Props{
    project: Project,
    deleteProject: (id: string) => void
}

class ProjectItem extends React.Component<Props> {

    constructor(props: any){
        super(props);

        this.deleteProject = this.deleteProject.bind(this);
    }    

    deleteProject(){
        let id = this.props.project.id;
        this.props.deleteProject(id)
    }

    render(){

        let {id, title} = this.props.project;

        return(
            <div>
                <div className="project_item row"> 
                    <div className="col">{title}</div>
                    <div className="col">
                        <Link to={`/details/${id}`} >details</Link>
                    </div>
                    <div className="col">
                        <Link to={`/issues/${id}`}>to project</Link>
                    </div>
                    <div className="col">
                        <button 
                        onClick={this.deleteProject}
                        className="del_button icon_button"
                        >
                        <FaRegTrashAlt />
                        </button>
                    </div>
                </div>
            </div>
        )
    }
}

export default ProjectItem;