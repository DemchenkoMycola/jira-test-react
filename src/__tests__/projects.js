import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import ProjectItem from '../components/projects/ProjectItem';
import ProjectsForm from '../components/projects/AddProjects';

describe('projects', () => {
    const store = createStore(() => ({}));

    test('should render project item', () => {

        const project = {
            "id": "1",
            "title": "project 1",
            "description": "description to project 1",
            "link": "http://google.com",
            "users": ["1", "2"]
          };

        const item = shallow(<ProjectItem project={project} />);

        expect(item).toMatchSnapshot();

    });

    test('should render project from', () => {
        const form = renderer.create(<ProjectsForm store={store} />)
        expect(form).toMatchSnapshot();
    });

});
