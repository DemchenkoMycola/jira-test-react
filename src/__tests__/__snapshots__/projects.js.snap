// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects should render project from 1`] = `
<div
  className="modal"
>
  <button
    className="button close_modal"
  >
    close
  </button>
  <form
    className="projects_form"
    onSubmit={[Function]}
  >
    <legend>
      Add project
    </legend>
    <label>
      Title
    </label>
    <input
      className="input"
      name="title"
      onChange={[Function]}
      placeholder="title"
      type="text"
    />
    <label>
      Description
    </label>
    <input
      className="input"
      name="description"
      onChange={[Function]}
      placeholder="description"
      type="text"
    />
    <label>
      Link to docs
    </label>
    <input
      className="input"
      name="link"
      onChange={[Function]}
      placeholder="link to docs"
      type="text"
    />
    <button
      disabled={true}
      type="submit"
    >
      add
    </button>
  </form>
</div>
`;
